File: 1500-count-largest-group/count-largest-group.java
class Solution {
    public int countLargestGroup(int n) {
        int[] digitSumCount = new int[37];
        int maxGroupSize = 0;

        for (int i = 1; i <= n; i++) {
            int sum = digitSum(i);
            digitSumCount[sum]++;
            maxGroupSize = Math.max(maxGroupSize, digitSumCount[sum]);
        }

        int result = 0;
        for (int count : digitSumCount) {
            if (count == maxGroupSize) {
                result++;
            }
        }

        return result;
    }

    private int digitSum(int num) {
        int sum = 0;

        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        
        return sum;
    }
}

File: 1500-count-largest-group/count-largest-group.go
func countLargestGroup(n int) int {
    digitSumCount := make([]int, 37);
    digitSum := make([]int, n + 1);
    maxGroupSize, result := 0, 0;

    digitSumCount[0] = -1;
    
    for i := 0; i <= n; i++ {
        digitSum[i] = i % 10 + digitSum[i / 10];
        digitSumCount[digitSum[i]]++;
    }

    for _, group := range digitSumCount {
        if maxGroupSize < group {
            maxGroupSize = group;
            result = 0;
        }

        if maxGroupSize == group {
            result++;
        }
    }

    return result;
}